core/src/main/java/org/elasticsearch/cluster/metadata/IndexGraveyard.java

```diff
@@ -0,0 +1,420 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.cluster.metadata;
+
+import org.elasticsearch.cluster.Diff;
+import org.elasticsearch.cluster.Diffable;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamInputReader;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.index.Index;
+
+import java.io.IOException;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Deque;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A collection of tombstones for explicitly marking indices as deleted in the cluster state.
+ *
+ * The cluster state contains a list of index tombstones for indices that have been
+ * deleted in the cluster.  Because cluster states are processed asynchronously by
+ * nodes and a node could be removed from the cluster for a period of time, the
+ * tombstones remain in the cluster state for a fixed period of time, after which
+ * they are purged.
+ */
+final public class IndexGraveyard implements ToXContent, Diffable<IndexGraveyard> {
+
+    private final List<Tombstone> tombstones;
+
+    private IndexGraveyard(final Collection<Tombstone> list) {
+        tombstones = Collections.unmodifiableList(new ArrayList<>(list));
```

Comment:
You are right, it was a remnant from when I was using `Deque` in the builder, and an unmodifiable list can only be created from `List` objects.  I just changed it.